pulp_ansible/app/galaxy/v1/tasks.py

```diff
@@ -0,0 +1,184 @@
+                alternate_role_name=alternate_role_name,
+            )
+            """
+            if gitrepo.tags:
+                github_reference = gitrepo.tags[-1].name
+
+        """
+        # if no tags, default to the commit hash?
+        # WARNING: many roles upstream have -no- versions.
+        if github_reference is None:
+            github_reference = gitrepo.head.commit.hexsha
+        """
+
+        # Upstream stores this, pulp_ansible does not.
+        # github_commit = get_tag_commit_hash(
+        #     clone_url, github_reference, checkout_path=checkout_path
+        # )
+
+        # Upstream stores this as a "release date", pulp_ansible does not.
+        # github_commit_date = get_tag_commit_date(
+        #     clone_url, github_reference, checkout_path=checkout_path
+        # )
+
+        # pulp_ansible does not have role tags
+        """
+        # parse out the role metadata
+        role_meta = get_path_role_meta(checkout_path)
+        role_tags = role_meta.get("galaxy_info", {}).get("galaxy_tags", [])
+        if role_tags is None:
+            role_tags = []
+        """
+
+        # pulp_ansible does not have a model or a view for readme content
+        """
+        ldr = RoleLoader(
+            content_type="role",
+            root=os.path.dirname(checkout_path),
+            rel_path=os.path.basename(checkout_path),
+        )
+        readme = ldr._get_readme()
+        readme_html = markup_utils._render_from_markdown(readme)
+        """
+
+        metadata = {
+            "github_user": github_user,
+            "github_repo": github_repo,
+            "role_name": role_name,
+            "name": role_name,
+            "namespace": github_user,
+            "role_version": github_reference,
+        }
+
+        # pulp_ansible's Roles are actuallly Role Versions
+        # The roles list viewset aggregates the versions together
+        # by using a {namespace}.{name} as the primary key.
+        role = Role(
+            version=metadata["role_version"],
+            name=metadata["name"],
+            namespace=metadata["namespace"],
+        )
+        role.save()
+
+        relative_path = "%s/%s/%s.tar.gz" % (
+            metadata["github_user"],
+            metadata["role_name"],
+            metadata["role_version"],
+        )
+
+        # Make a tarball
+        shutil.rmtree(os.path.join(checkout_path, ".git"))
+        tarfn = f"/tmp/{github_user}.{role_name}.{github_reference}.tar.gz"
+        cmd = f"tar czvf {tarfn} {checkout_path}"
+        pid = subprocess.run(cmd, shell=True)
+        assert pid.returncode == 0
+
+        # Create a real artifact
+        artifact = Artifact.init_and_validate(tarfn)
+        artifact.save()
+
+        # This is a pulp'ism for the source url
+        # but we will generate the download_url in the
+        # serializer so this wouldn't ever be
+        # used ... ?
+        relative_path = "%s/%s/%s.tar.gz" % (
```

Comment:
I believe the content app needs this at serving time, otherwise it can't know. Internally the files aren't stored with filenames like `something.tar.gz` it's stored in a sha256 content addressable storage. I suspect (maybe) if this wasn't right it would not work?